modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_builder.py

```diff
@@ -377,11 +377,41 @@ def gen_reduce_expr(self):
                 self._dtype,
             )
 
+    class QuantileAggregate(CompoundAggregateWithColArg):
+        """
+        A QUANTILE aggregate generator.
+
+        Parameters
+        ----------
+        builder : CalciteBuilder
+            A builder to use for translation.
+        arg : List of BaseExpr
+            An aggregated values.
+        """
+
+        def __init__(self, builder, arg):
```

Comment:
Looks easier to read
```suggestion
        def __init__(self, builder, arg):
            dtype = arg[0]._dtype if is_datetime64_dtype(arg[0]._dtype) else _get_dtype("float64")
```